Src/Support/Google.Apis/Requests/BatchRequest.cs

```diff
@@ -212,10 +215,23 @@ public async Task ExecuteAsync(CancellationToken cancellationToken)
                 {
                     // Parse the current content object.
                     var responseContent = await responseMessage.Content.ReadAsStringAsync().ConfigureAwait(false);
-                    var content = service.Serializer.Deserialize(responseContent,
-                        allRequests[requestIndex].ResponseType);
+                    object deserializedContent = responseContent;
+                    RequestError error = null;
+                    try
+                    {
+                        deserializedContent = service.Serializer.Deserialize(responseContent,
+                            allRequests[requestIndex].ResponseType);
+                    }
+                    catch(Exception ex)
+                    {
+                        error = new RequestError
+                        {
+                            Message = $"The response was read but could not be deserialized using the {nameof(service.Serializer)}.",
+                            ErrorResponseContent = ex.ToString()
```

Comment:
Why is this `ex.ToString` rather than `responseContent`? I may have misunderstood something.

Reply:
This is awful, but:

- The request here succeeded, the response content is not an error, is just in a format that we don't know how to deserialize.
- We are already sending the string content on the `content` parameter of the callback, so I figured it'd be more useful to send here the exception content.
- There's no way in which we can send the Exception object to the callback, except for throwing here of course, which stops the whole processing. So even if only one request was problematic, we don't process anything else after.

I'll create and alternative commit that is less weird maybe. And we can choose between those two.

Reply:
Ack.